File: backend/src/main/java/com/app_financiera/api/util/PasswordPolicy.java
package com.app_financiera.api.util;

public final class PasswordPolicy {

    private PasswordPolicy() {
    }

    /**
     * Reglas para restablecer contraseña (más estrictas que el registro inicial).
     */
    public static String validateForReset(String password) {
        if (password == null || password.isBlank()) {
            return "La contraseña es requerida";
        }
        if (password.length() < 10) {
            return "La contraseña debe tener al menos 10 caracteres";
        }
        if (!password.chars().anyMatch(Character::isUpperCase)) {
            return "La contraseña debe contener al menos una letra mayúscula";
        }
        if (!password.chars().anyMatch(Character::isLowerCase)) {
            return "La contraseña debe contener al menos una letra minúscula";
        }
        if (!password.chars().anyMatch(Character::isDigit)) {
            return "La contraseña debe contener al menos un dígito";
        }
        if (!password.matches(".*[^A-Za-z0-9].*")) {
            return "La contraseña debe contener al menos un carácter especial";
        }
        return null;
    }
}

File: frontend/src/lib/password-policy.ts
/** Debe coincidir con PasswordPolicy.validateForReset del backend */
export function validateResetPassword(password: string): string | null {
  if (!password) return 'La contraseña es requerida';
  if (password.length < 10) return 'La contraseña debe tener al menos 10 caracteres';
  if (!/[A-Z]/.test(password)) return 'La contraseña debe contener al menos una letra mayúscula';
  if (!/[a-z]/.test(password)) return 'La contraseña debe contener al menos una letra minúscula';
  if (!/\d/.test(password)) return 'La contraseña debe contener al menos un dígito';
  if (!/[^A-Za-z0-9]/.test(password)) return 'La contraseña debe contener al menos un carácter especial';
  return null;
}

export const resetPasswordChecks = (password: string) => ({
  minLength: password.length >= 10,
  hasUpper: /[A-Z]/.test(password),
  hasLower: /[a-z]/.test(password),
  hasNumber: /\d/.test(password),
  hasSpecial: /[^A-Za-z0-9]/.test(password),
});
